tests/phpunit/includes/Core/Modules/Module_With_Owner_ContractTests.php

```diff
@@ -0,0 +1,32 @@
+<?php
+/**
+ * Module_With_Owner_ContractTests
+ *
+ * @package   Google\Site_Kit\Tests\Core\Modules
+ * @copyright 2019 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Modules;
+
+use Google\Site_Kit\Core\Modules\Module_With_Owner;
+use Google\Site_Kit\Tests\TestCase_Context_Trait;
+
+trait Module_With_Owner_ContractTests {
+
+	use TestCase_Context_Trait;
+
+	/**
+	 * @return Module_With_Owner
+	 */
+	abstract protected function get_module_with_owner();
+
+	public function test_owner_id() {
+		$testcase = $this->get_testcase();
+		$module   = $this->get_module_with_owner();
+
+		// By default, ownerID is 0.
+		$testcase->assertEquals( 0, $module->get_owner_id() );
+	}
+}
```

Comment:
Let's add another test or two here to cover a non-zero return value as well. To do that, we'll need to reach slightly beyond the `Module_With_Owner` interface alone, but that's okay as it's all within the context of the implicit contract.

In order to test this, we can assert that `$module->get_settings()` is an instance of `Setting_With_Owned_Keys_Interface`. We need to register the settings and then we can call `$settings->get_owned_keys()` and change one of them with the current user set to an admin with permissions. Then we can check that `$module->get_owner_id()` returns the expected user ID.

There isn't really a cleaner way to set the module's owner ID, but this makes the test a little less fragile to change in the future.

We don't need to be as thorough here and test each owned key and all the different possible combinations of permissions here as that's all well tested in the other tests. We just want to trigger the value to change and to ensure the new value comes back as expected.

Reply:
Actually, i am not sure that it is going to work at all because Module_With_Owner_Trait uses Owner_ID setting that pulls owner ID from `googlesitekit_owner_id` and has nothing to do with ownerID from module settings. I think we don't need `Module_With_Owner_Trait` or we need to make it work differently.

Reply:
Ah, I see what you mean; I missed that in my initial review 😞 . The `Module_With_Owner_Trait::get_owner_id` method should return the `ownerID` from the module's settings as described in the IB – not the `googlesitekit_owner_id` which is the "site" owner and may be different than a module's owner. 

Regarding the `Module_With_Owner_Trait`, this is a bit complicated however considering the flexibility of the key used for holding the `ownerID`. `get_owner_id` essentially needs to return `$this->get_settings()->get()['ownerID']`.

@felixarntz I think perhaps we should remove the `$owner_id_key` parameter to `register_owned_keys` as it makes it complicated for other parts to account for this variable key and we don't really need it for the time being. What do you think?

Reply:
@aaemnnosttv I originally suggested introducing that key, but that's a fair point. In combination with my comment further above, this would furthermore allow us to simplify `register_owned_keys` as a parameter-less method, so +1 from my end.

Reply:
Okay, it should be good now.